source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -0,0 +1,795 @@
+    }
+
+    template <int dim>
+    double
+    EllipsoidalChunk<dim>::get_rotation_longitude()
+    {
+      return rotation_longitude;
+    }
+
+    template <int dim>
+    double
+    EllipsoidalChunk<dim>::get_rotation_latitude()
+    {
+      return rotation_latitude;
+    }
+
+    template <int dim>
+    double
+    EllipsoidalChunk<dim>::length_scale() const
+    {
+      // diameter divided by 20
+      return ((xyz_from_lat_long_height_in_ellipsoid(southLatitude,
+                                                     eastLongitude,
+                                                     -bottomDepth)//,Radius,Eccentricity)
+               - xyz_from_lat_long_height_in_ellipsoid(northLatitude, westLongitude, 0/*,Radius,Eccentricity*/)).norm()
+              / 20);
+    }
+
+    template <int dim>
+    Point<dim>
+    EllipsoidalChunk<dim>::representative_point(const double depth) const
+    {
+      return Point<dim>();
+    }
+
+    template <>
+    Point<3>
+    EllipsoidalChunk<3>::representative_point(const double depth) const
+    {
+      const int dim = 3;
+      Assert(depth >= 0, ExcMessage("Given depth must be positive or zero."));
+      Assert(depth <= maximal_depth(),
+             ExcMessage("Given depth must be less than or equal to the maximal depth of this geometry."));
+
+      // choose a point on the center axis of the domain
+      Point<dim> p =
+        (xyz_from_lat_long_height_in_ellipsoid(southLatitude,
+                                               eastLongitude,
+                                               -bottomDepth)
+         + xyz_from_lat_long_height_in_ellipsoid(northLatitude, eastLongitude, 0))
+        / 2;
+      p /= p.norm();
+      p *= get_radius(p) - depth;
+      return p;
+    }
+
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace GeometryModel
+  {
+    ASPECT_REGISTER_GEOMETRY_MODEL(EllipsoidalChunk,
+                                   "ellipsoidal chunk",
+                                   "A 3D chunk geometry that accounts to Earth's ellipticity (default assuming the "
+                                   "WGS84 ellipsoid definition) which can be defined in non-coordinate directions. "
+                                   "The user has two options for creating a ellipsoidal chunk geometry: 1) by defining "
+                                   "two opposing points (SW and NE or NW and SE) a coordinate parallel ellipsoidal "
+                                   "chunk geometry will be created. 2) by defining three points a non-coordinate "
+                                   "parallel ellipsoidal chunk will be created. The points are defined in the input "
+                                   "file by longitude:latitude. It is also possible to define additional subdivisions "
+                                   "in each direction. Faces of model are defined as 0, west; 1,east; 2, south; 3, "
+                                   "north; 4, bottom; 5, top. ")

```

Comment:
This is mostly for my own curiosity: I assume that this enumeration of boundary sides assumes that the user really put the northeast point northeast of the southwest point, right? If you don't, then these names aren't going to be for boundaries that really match the names. This is not a problem, I'm just curious to hear what happens in that case.